modules/heat_conduction/src/fvbcs/FunctorThermalResistanceBC.C

```diff
@@ -0,0 +1,177 @@
+                                            "emissivity of the surface");
+
+  params.addRequiredParam<std::vector<Real>>(
+      "thermal_conductivities",
+      "vector of thermal conductivity values used for the conduction layers");
+  params.addRequiredParam<std::vector<Real>>("conduction_thicknesses",
+                                             "vector of conduction layer thicknesses");
+
+  MooseEnum geometry("cartesian cylindrical", "cartesian");
+  params.addParam<MooseEnum>("geometry", geometry, "type of geometry");
+  params.addRangeCheckedParam<Real>("inner_radius",
+                                    "inner_radius > 0.0",
+                                    "coordinate corresponding to the first resistance layer");
+
+  params.addRangeCheckedParam<Real>(
+      "step_size", 0.1, "step_size > 0.0", "underrelaxation step size");
+
+  params.addRangeCheckedParam<unsigned int>(
+      "max_iterations", 100, "max_iterations >= 0", "maximum iterations");
+
+  params.addRangeCheckedParam<Real>(
+      "tolerance", 1E-3, "tolerance > 0.0", "tolerance to converge iterations");
+  params.addClassDescription("Thermal resistance heat flux boundary condition for the "
+                             "fluid and solid energy equations");
+  return params;
+}
+
+FunctorThermalResistanceBC::FunctorThermalResistanceBC(const InputParameters & parameters)
+  : FVFluxBC(parameters),
+    _geometry(getParam<MooseEnum>("geometry").getEnum<Moose::CoordinateSystemType>()),
+    _inner_radius(
+        _geometry == Moose::CoordinateSystemType::COORD_RZ ? getParam<Real>("inner_radius") : 1.0),
+    _T(isParamValid("temperature") ? getFunctor<ADReal>("temperature")
+                                   : getFunctor<ADReal>("variable")),
+    _T_ambient(getParam<Real>(HeatConduction::T_ambient)),
+    _k(getParam<std::vector<Real>>("thermal_conductivities")),
+    _dx(getParam<std::vector<Real>>("conduction_thicknesses")),
+    _h(getFunctor<ADReal>(getParam<MooseFunctorName>("htc"))),
+    _emissivity(getParam<Real>(HeatConduction::emissivity)),
+    _max_iterations(getParam<unsigned int>("max_iterations")),
+    _tolerance(getParam<Real>("tolerance")),
+    _alpha(getParam<Real>("step_size")),
+    _T_surface(0.0),
+    _outer_radius(_inner_radius),
+    _conduction_resistance(0.0),
+    _parallel_resistance(0.0)
+{
+  if (_k.size() != _dx.size())
+    mooseError("Number of specified thermal conductivities must match the number "
+               "of conduction layers!");
+
+  if (_geometry == Moose::CoordinateSystemType::COORD_RZ)
+    for (const auto & d : _dx)
+      _outer_radius += d;
+
+  // because the thermal conductivities are constant, we only need to compute
+  // the conduction resistance one time
+  computeConductionResistance();
+}
+
+void
+FunctorThermalResistanceBC::computeConductionResistance()
+{
+  Real r = _inner_radius;
+
+  for (std::size_t i = 0; i < _k.size(); ++i)
```

Comment:
```suggestion
  for (const auto i : index_range(_k))
```

Reply:
ok I ll fix it in the other non-functor too